includes/class-payment-gateways.php

```diff
@@ -37,6 +48,86 @@ public function load_gateways() {
 		}
 	}
 
+	/**
+	 * Include Authorize.Net, only if it is already configured.
+	 * There is a filter as well, so it can also be enabled using a code snippet.
+	 *
+	 * @since x.x
+	 *
+	 * @return bool
+	 */
+	private function should_include_authorize_net_gateway() {
+		$settings = get_option( 'wpbdp_settings' );
+		if ( ! is_array( $settings ) ) {
+			$include_authorize_net = false;
+		} else {
+			$include_authorize_net = $this->settings_key_exists(
+				array(
+					'authorize-net-login-id',
+					'authorize-net-transaction-key',
+				)
+			);
+		}
+
+		/**
+		 * Allow flexibility so users can still opt into Authorize.Net even though it is hidden by default.
+		 *
+		 * @since x.x
+		 *
+		 * @param bool $include_authorize_net Whether to include the Authorize.Net gateway.
+		 */
+		return (bool) apply_filters( 'wpbdp_include_authorize_net_gateway', $include_authorize_net );
+	}
+
+	/**
+	 * Add Stripe by default, if the module is not already configured.
+	 * Avoid including the Stripe Lite module if the Stripe module is already configured.
+	 * This is to avoid Stripe conflicts with Stripe Lite when the Stripe module is active.
+	 *
+	 * @since x.x
+	 *
+	 * @return bool
+	 */
+	private function should_include_stripe_lite_gateway() {
+		if ( ! class_exists( 'WPBDP__Stripe__Gateway' ) ) {
+			// Always include if the BD module is not installed.
+			return true;
+		}
+
+		return ! $this->settings_key_exists(
+			array(
+				'stripe-test-publishable-key',
+				'stripe-test-secret-key',
+				'stripe-live-publishable-key',
+				'stripe-live-secret-key',
+			)
+		);
+	}
+
+	/**
+	 * Check if at least one of the keys specified has a value in settings.
+	 *
+	 * @since x.x
+	 *
+	 * @param array $keys
+	 *
+	 * @return bool
+	 */
+	private function settings_key_exists( $keys ) {
+		$settings = get_option( 'wpbdp_settings' );
+		if ( ! is_array( $settings ) ) {
+			return false;
+		}
+
+		foreach ( $keys as $key ) {
+			if ( ! empty( $settings[ $key ] ) ) {
```

Comment:
We really only want to know if one key exists for the purpose of this check.

It's really just to determine if we should hide their settings or not. If something/anything is set, we don't want to hide it.

Reply:
Understood!

I think we could add it as a public method in [this settings class](https://github.com/Strategy11/business-directory-plugin/blob/master/includes/admin/settings/class-settings.php). It would make sense and it would also be easy to use through the `$wpbdp` global.

Something like this:
```php
$wpbdp->settings->any_setting_exists( $keys );
```

The current functions are used twice in `class-payment-gateways.php` and the other one is used only once in `class-modules.php`.

Reply:
Thank you @sorinmarta! That's a great improvement.

Updated in https://github.com/Strategy11/business-directory-plugin/pull/402/commits/f16ad78bf9ac2980e1c8b38f4f8ed6b9bb400518

And also https://github.com/Strategy11/business-directory-plugin/pull/402/commits/088aecd2dd19965d2ec677e7a87772b1181d04a6